### ISSUE TITLE
Rust toolchain and musl target

### ISSUE THREAD

**Issue by Toorero** [LukeMathWalker/cargo-chef#279]
I can't build the musl target `x86_64-unknown-linux-musl` because it is not installed by the toolchain specified in `rust-toolchain.toml`.

My current workaround is the following in the builder
```docker
cargo chef cook --no-build --release --target x86_64-unknown-linux-musl --recipe-path recipe.json                                                     
RUN rustup target add x86_64-unknown-linux-musl                                                                                                           
RUN cargo chef cook --release --target x86_64-unknown-linux-musl --recipe-path recipe.json      
```

I suspect the problem is that the target is installed via `rustup target add ...` before the `rust-toolchain.toml` is installed, but even manually copying the `rust-toolchain.toml` to the container doesn't seem to help so maybe something else is failing.

**Comment by LukeMathWalker** [comment]
Can you provide a reproducible example as a repository for us to investigate?

**Comment by Toorero** [comment]
Wait let me see if the repositories I'm using are all public. Then I can just share you this.

**Comment by Toorero** [comment]
Sure: [Dockerfile](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord.eBPF.broken).

I also included stuff I tried like generally copying the `rustup-toolchain.toml` but without any effect.
For me the file reliably fails at [line 20](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord.eBPF.broken#L20). The [working version](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord.eBPF) has the aforementioned [workaround](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord.eBPF#L17-19) applied

**Comment by LukeMathWalker** [comment]
That seems to be working as expected.

If you're cross-compiling, it's on you to install the target toolchain. `cargo-chef` won't do it for you.

You can get `rustup` to do it automatically, in its current latest version, by copying the `rust-toolchain.toml` file and then invoking `rustup show`. This allows you to avoid duplicating the target across the toolchain file and the Dockerfile.

Nothing wrong on the `cargo-chef` side though.

**Comment by Toorero** [comment]
> If you're cross-compiling, it's on you to install the target toolchain. cargo-chef won't do it for you.

But even doing [this](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord.eBPF.broken#L18) before running cargo-chef doesn't seem to work although [this](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord#L13) works in the other images.

**Comment by LukeMathWalker** [comment]
> > If you're cross-compiling, it's on you to install the target toolchain. cargo-chef won't do it for you.
> 
> But even doing [this](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord.eBPF.broken#L18) before running cargo-chef doesn't seem to work although [this](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/docker/Dockerfile.supervisord#L13) works in the other images (where you don't need the nightly toolchain)

Why should `cargo-chef` use `nightly` in the first Dockerfile?

**Comment by Toorero** [comment]
Because it is required if I want to compile using [aya](https://aya-rs.dev/) and I specified so in the [`rust-toolchain.toml`](https://gitlab.kit.edu/kit/tm/telematics/kira/kira-rust/-/blob/fix/cargo-chef/rust-toolchain.toml).

**Comment by Toorero** [comment]
I think in general the problem still is `cargo chef cook` is not respecting the however present `rust-toolchain.toml` even thoug it is collected when running `cargo chef prepare`.

After running `cargo chef cook  --release --target x86_64-unknown-linux-musl --recipe-path recipe.json` the toolchain file is present but I suspect only after chef started compiling the dependencies. Still an issue impov.

**Comment by Toorero** [comment]
Ok, I think I'm going to implement a minimum (not) working example and a docker file how I think it should work out of the box without the additional commands needed.

**Comment by Toorero** [comment]
Ok here is the [mwe](https://github.com/Toorero/cargo-chef/tree/mwe/279/mwe)

**Comment by Toorero** [comment]
The [Dockerfile](https://github.com/Toorero/cargo-chef/blob/mwe/279/mwe/Dockerfile) is the "workaround" for both issues.
